crates/autopilot/src/limit_order_quoter.rs

```diff
@@ -9,6 +9,22 @@ use model::{
 use shared::order_quoting::{OrderQuoting, QuoteParameters};
 use std::sync::Arc;
 
+#[derive(prometheus_metric_storage::MetricStorage, Clone, Debug)]
+#[metric(subsystem = "limit_order_quoter")]
+struct Metrics {
+    /// Histogram for counting ignored limit orders.
+    ignored: prometheus::Histogram,
```

Comment:
I'm not sure that a `Histogram` is the best tool for these metrics. They collect data in buckets which have to be configured appropriately. If requirements for the buckets change the code would have to be adjusted as well.
AFAIK the default buckets are `0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1.0, 2.5, 5.0, 10.0`.
I think we could also get useful metrics out of an `IntCounter` without having to tweak the buckets in the future when more and more orders will get filtered out because they are out of the market price.

Reply:
We've had similar problems in some Driver metric. Prometheus samples metrics in intervals like 15 or 30 seconds. If you have a process that runs every second and you export the current state as a metric then prometheus will only catch it every nth time. Alternatively you can use a histogram to catch all values but lose some granularity. Another consideration is a histogram with no buckets which is a total sum and a counter for how often it happened. With that you can still have a meaningful average over time intervals.

I'm not immediately sure what is best here to make the metrics meaningful. Imagining this from the perspective of what I'd want to see in Grafana I think what we want is a current state counter on the total number of limit orders in the system and the total number of them considered in the current auction. We care about what is in the auction and not that much what happens in the background task.

Reply:
Could have a collection of current state (Gauge) metrics like:
- number of unexpired limit orders
- number of unexpired limit orders missing quotes
- number of unexpired limit orders having quotes
- number of unexpired limit orders excluded due to heuristics

Some of these would execute a new DB query to calculate them.

Reply:
> I can see how the other ones would be calculated, but it seems to me that the correct place to update these metrics is during auction cutting, is that correct? I can do that in a follow-up PR.

Agreed.